onnx2kerastl/elementwise_layers.py

```diff
@@ -71,33 +72,71 @@ def convert_elementwise_add(node, params, layers, lambda_func, node_name, keras_
     logger = logging.getLogger('onnx2keras.add')
 
     if len(node.input) != 2:
-        raise AttributeError('Number of inputs is not equal 2 for element-wise layer')
+        raise AttributeError('Number of inputs is not equal to 2 for element-wise layer')
 
     input_0 = layers[node.input[0]]
     input_1 = layers[node.input[1]]
 
-    input_0_is_non_keras = is_numpy(input_0) or isinstance(input_0, EagerTensor)
-    input_1_is_non_keras = is_numpy(input_1) or isinstance(input_1, EagerTensor)
+    input_0_is_constant = is_numpy(input_0) or isinstance(input_0, EagerTensor)
+    input_1_is_constant = is_numpy(input_1) or isinstance(input_1, EagerTensor)
+
     try:
-        if not input_0_is_non_keras and not input_1_is_non_keras:
-            to_add = input_1
-            # We probably need to seperate two possibilities here. Currently we only deal with second option
-            # [Batch] + [Batch,1] -> [Batch,1]
-            # [Not-Batch] + [Not,Batch,1] -> [Not-batch, Not-batch]
+        if not input_0_is_constant and not input_1_is_constant:
+            # Both inputs are variables
             if len(input_0.shape) != len(input_1.shape):
-                layers[node_name] = tf_add(input_0, to_add, tf_name=f"{params['cleaned_name']}_add")
+                # Use TensorFlow add to handle shape differences
+                layers[node_name] = tf_add(input_0, input_1, tf_name=f"{params['cleaned_name']}_add")
             else:
-                layers[node_name] = keras.layers.Add(name=f"{params['cleaned_name']}_add")([input_0, to_add])
+                # Use Keras Add layer
+                layers[node_name] = keras.layers.Add(name=f"{params['cleaned_name']}_add")([input_0, input_1])
         else:
             raise ValueError('Operands are different.')
     except (IndexError, ValueError):
-        logger.warning('Failed to use keras.layers.Add. Fallback to TF lambda.')
-        if input_0_is_non_keras:
-            layers[node_name] = input_1 + input_0
+        logger.warning('Failed to use keras.layers.Add. Fallback to Lambda layer.')
+
+        if input_0_is_constant and not input_1_is_constant:
+            # input_0 is constant, input_1 is variable
+            constant_value = np.asarray(tf.cast(input_0, dtype=input_1.dtype))
+            variable_input = input_1
+
```

Comment:
Can we end the if condition here and then take the next part and combine it to avoid code duplication?